Src/HALALMock/Services/Communication/SPI/SPI.cpp

```diff
@@ -512,19 +436,22 @@ void HAL_SPI_ErrorCallback(SPI_HandleTypeDef* hspi) {
 void SPI::spi_communicate_order_data(SPI::Instance* spi, uint8_t* value_to_send,
                                      uint8_t* value_to_receive,
                                      uint16_t size_to_send) {
-    HAL_SPI_TransmitReceive_DMA(spi->hspi, value_to_send, value_to_receive,
-                                size_to_send);
-}
+    // Search for the id of the SPI instance
+    uint8_t id;
+    for (auto iter : registered_spi) {
```

Comment:
use structure bindings

Reply:
What do you mean?

Reply:
I'm guessing [this](https://en.cppreference.com/w/cpp/language/structured_binding)

Reply:
Done